interfaces/apparmor/apparmor.go

```diff
@@ -64,33 +64,70 @@ const (
 	skipReadCache aaParserFlags = 1 << iota
 )
 
-func loadProfile(fname, cacheDir string, flags aaParserFlags) error {
+func loadProfiles(fnames []string, cacheDir string, flags aaParserFlags) error {
+	if len(fnames) == 0 {
+		return nil
+	}
+
 	// Use no-expr-simplify since expr-simplify is actually slower on armhf (LP: #1383858)
-	args := []string{"--replace", "--write-cache", "-O", "no-expr-simplify", fmt.Sprintf("--cache-loc=%s", cacheDir)}
+	args := []string{"--replace", "--write-cache", "-O", "no-expr-simplify",
+		fmt.Sprintf("--cache-loc=%s", cacheDir)}
 	if flags&skipReadCache != 0 {
 		args = append(args, "--skip-read-cache")
 	}
 	if !osutil.GetenvBool("SNAPD_DEBUG") {
 		args = append(args, "--quiet")
 	}
-	args = append(args, fname)
+	args = append(args, fnames...)
 
 	output, err := exec.Command("apparmor_parser", args...).CombinedOutput()
 	if err != nil {
-		return fmt.Errorf("cannot load apparmor profile: %s\napparmor_parser output:\n%s", err, string(output))
+		return fmt.Errorf("cannot load apparmor profiles: %s\napparmor_parser output:\n%s", err, string(output))
+	}
+	return nil
+}
+
+func reloadProfiles(profiles []string, profileDir, cacheDir string) error {
```

Comment:
Sorry to bother you with this Alfonso, as I realize this was already a problem in the existing code, but maybe it's time to polish it a bit.

There's nothing about reload in the implementation of this function below. This is simply calling loadProfile. Can we please get rid of it and call the real loadProfiles at the call sites?  The only thing being done here is adding a filename prefix to the filename. We shouldn't have that logic inside this function. It's error prone to have that difference between load and reload, and it also seems error prone to have a reload that does nothing more than load.